students/hirot_directory/fib_hiroTakechi.py

```diff
@@ -0,0 +1,53 @@
+
+'''
+Fibonacchi recursion Python code
+
+'''
+
+# in-class:
+
+def fib(n):
+
+	if n == 0:
+		return 0
+	if n == 1:
+		return 1
+	return fib(n-1) + fib(n-2)
+
+if __name__ == '__main__':
+	assert fib(0) == 0
+	assert fib(1) == 1
+	assert fib(2) == 1
+	assert fib(3) == 2
+	print(fib(10))
+
+# the codes that I worked on after class: 
+
+## code 1:
+def fibonacci(n, a=1, b=0):
```

Comment:
this is VERY slick! I've never seen this solution, but it's efficient and clean. Though kinda hard to follow the logic.

Reply:
Hi Chris,

I think this is from the recursion exercise (or homework) of Foundations of Algorithm and Data Structure, if I remember correctly. 

Thanks,
Hiro

Reply:
you could have claimed credit :-)